samples/bluetooth/peripheral_hids_keyboard/src/main.c

```diff
@@ -851,10 +851,13 @@ static void num_comp_reply(bool accept)
 
 static void button_changed(uint32_t button_state, uint32_t has_changed)
 {
+	static bool pairing_button_pressed;
 
 	uint32_t buttons = button_state & has_changed;
 
 	if (k_msgq_num_used_get(&mitm_queue)) {
+		pairing_button_pressed = true;
```

Comment:
Shouldn't this be set in the two individual if's below? Could be that none of them are pressed?

Reply:
It could be because there can be a timeout when you press button to late. I moved it to the individual scope also I a little improved the paring queue logic when paring timeout occurs then the item will be removed from queue.